### DISCUSSION TITLE
UX Goal: Support online teaching through IIIF A/V w/ text and image based annotations

### DISCUSSION THREAD

**Discussion by mejackreed** [ProjectMirador/mirador#3459] [Show and tell]
Desscribed by @knagasaki as a goal to support online teaching through using Mirador 3 and IIIF content. Currently (v3.0.0) Mirador only supports displaying `<audio>` and `<video>` without annotations.

Annotations could by text or image based to provide learning through watching the video.

An authoring tool is also needed for creating these types of annotations. YouTube does not provide this type of functionality.

@knagasaki if you are able to provide a still of an image as an example this would be useful.

**Comment by 2SC1815J** [comment]
I developed a proof of concept version of Mirador 3 that supports displaying `<video>` with annotations.
This development is based on a request from @knagasaki, his organization, and several organizations for the cooperative work.
https://dh.l.u-tokyo.ac.jp/activity/iiif/video-annotation

Demo:
https://dzkimgs.l.u-tokyo.ac.jp/videos/m3/embed.html?iiif-content=https://dzkimgs.l.u-tokyo.ac.jp/videos/iiif_in_japan_2017/manifest.json
https://dzkimgs.l.u-tokyo.ac.jp/videos/m3/embed.html?iiif-content=https://dzkimgs.l.u-tokyo.ac.jp/videos/cat2020/manifest.json

Currently, Mirador 3 does not support displaying a mixture of audios, videos, and still images.
This time, I have added support for displaying annotations on videos with minimal changes to the existing implementation.
The following limitations exist:
- Only one video can be displayed at a particular point in time.
- If the Canvas size is different from the video size, or if the video is associated with some space dimensions on the Canvas, the annotations may not be displayed in the correct position.

We consider the version we have developed to be a useful "throw-away" implementation until a more extensive and fundamental reimplementation takes place.

The source code will be available later.

Have happy holidays!
